src/asqi/validation.py

```diff
@@ -2,12 +2,91 @@
 from pathlib import Path
 from typing import Any, Dict, List, Optional
 
+import yaml
 from pydantic import BaseModel
+from rich.console import Console
 
+from asqi.config import load_config_file
+from asqi.errors import DuplicateTestIDError
 from asqi.schemas import Manifest, SuiteConfig, SystemsConfig
 
 logger = logging.getLogger()
 
+# Initialize Rich console and execution queue
+console = Console()
+
+
+def validate_test_ids(test_suite_config_path: str) -> None:
+    """
+    Validates that all test IDs in the suite file are unique.
+
+    Args:
+        test_suite_config_path: Path to the test suite file
+
+    Raises:
+        DuplicateTestIDError: If duplicate test IDs are found, with detailed information about all duplicate locations
+    """
+    all_test_ids: Dict[str, List[str]] = {}
+
+    if test_suite_config_path:
+        try:
+            suite_config = load_config_file(test_suite_config_path)
+            extract_suite_ids(all_test_ids, suite_config, test_suite_config_path)
+
+        except yaml.YAMLError as e:
+            console.print(
+                f"[yellow]ID Validation. Failed to parse test suite file {test_suite_config_path}:[/yellow] {e}"
+            )
+        except Exception as e:
+            console.print(
+                f"[yellow]IF Validation. Failed to process test suite file {test_suite_config_path}:[/yellow] {e}"
+            )
+    duplicate_dict = get_duplicate_test_ids(all_test_ids)
+    if duplicate_dict:
+        raise DuplicateTestIDError(duplicate_dict)
+
+
+def extract_suite_ids(
+    all_test_ids: Dict[str, List[str]],
+    suite_config: Dict[str, Any],
+    test_suite_config_path: str,
+) -> None:
+    """
+    Extract test IDs from a suite config file and register them to the ID state.
+
+    Args:
+        all_test_ids: Test ID state
+        suite_config: Test suite config dict
+        test_suite_config_path: Path to the test suite file
+    """
+
+    suite_name = suite_config.get("suite_name", "Unknown Suite")
+    test_suite = suite_config.get("test_suite", [])
+    for test in test_suite:
+        test_id = test["id"]
+        test_name = test["name"]
```

Comment:
access `id` like this is fine and `KeyError` will be raised if not available. But, if test_name is now optional we should use `.get` instead with a fallback name?